src/dappsExplorer/useOpenDapp.tsx

```diff
@@ -22,6 +22,8 @@ const useOpenDapp = () => {
   const [selectedDapp, setSelectedDapp] = useState<ActiveDapp | null>(null)
   const dispatch = useDispatch()
 
+  const extraAnalyticsPropertiesRef = useRef<Record<string, string | number>>({})
```

Comment:
this is a bit hacky but i've left it here because i think this hook can be removed soon if we remove the confirmation bottom sheet. i haven't been able to confirm this with Ellen yet so i've left it here but i haven't optimised the solution for this bottom sheet for sure...

Reply:
Ah another reason this "return component from hook" pattern feels off to me if we have to resort to this 😅